tests/vm_networking_test.go

```diff
@@ -89,61 +89,51 @@ var _ = Describe("Networking", func() {
 	tests.BeforeAll(func() {
 		tests.BeforeTestCleanup()
 
-		inboundChan := make(chan *v1.VirtualMachine)
-		outboundChan := make(chan *v1.VirtualMachine)
+		// Create and start inbound VM
+		inboundVM = tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+		inboundVM.Labels = map[string]string{"expose": "me"}
+		inboundVM.Spec.Subdomain = "myvm"
+		inboundVM.Spec.Hostname = "my-subdomain"
+		_, err = virtClient.VM(tests.NamespaceTestDefault).Create(inboundVM)
+		Expect(err).ToNot(HaveOccurred())
 
-		createAndLogin := func(labels map[string]string, hostname string, subdomain string) (vm *v1.VirtualMachine) {
-			vm = tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
-			vm.Labels = labels
-			vm.Spec.Subdomain = subdomain
-			vm.Spec.Hostname = hostname
+		// Create and start outbound VM
+		outboundVM = tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+		_, err = virtClient.VM(tests.NamespaceTestDefault).Create(outboundVM)
+		Expect(err).ToNot(HaveOccurred())
 
-			// Start VM
-			vm, err = virtClient.VM(tests.NamespaceTestDefault).Create(vm)
-			Expect(err).ToNot(HaveOccurred())
-			tests.WaitForSuccessfulVMStartIgnoreWarnings(vm)
+		for _, networkVm := range []*v1.VirtualMachine{inboundVM, outboundVM} {
+			// Wait for VM start
+			tests.WaitForSuccessfulVMStart(networkVm)
 
 			// Fetch the new VM with updated status
-			vm, err = virtClient.VM(tests.NamespaceTestDefault).Get(vm.ObjectMeta.Name, v13.GetOptions{})
+			vm, err := virtClient.VM(tests.NamespaceTestDefault).Get(networkVm.Name, v13.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
 			// Lets make sure that the OS is up by waiting until we can login
 			expecter, err := tests.LoggedInCirrosExpecter(vm)
 			defer expecter.Close()
 			Expect(err).ToNot(HaveOccurred())
```

Comment:
because I see it: a `defer` must be done after the err checks. Return values can be nil in case of errors.